core/src/main/java/org/wso2/msf4j/internal/router/HttpMethodInfo.java

```diff
@@ -323,4 +323,12 @@ public void end() throws Exception {
     public boolean isStreamingSupported() {
         return httpStreamHandler != null;
     }
+
+    public Object[] getArgs() {
+        return Arrays.copyOf(args, args.length);
```

Comment:
I had originally used without copying. But the problem was args[] has a "final" modifier but if we do not use array.copy(), internal objects can be modified by outside.